### PR TITLE
Feat  / add position management for perpetual strategy

### PR DESCRIPTION
**Pull request by vic-en**
**Before submitting this PR, please make sure**:

- [x] Your code builds clean without any errors or warnings
- [x] You are using approved title ("feat/", "fix/", "docs/", "refactor/")

**After creating this PR, please make sure:**

- [x] You link the related GitHub issue or ticket

**A description of the changes proposed in the pull request**:

### CODE REVIEW

**Review by Nullably — DISMISSED** [review]
LGTM

**Comment by RC-13** [comment]
Tested PR and so far only one minor issue was seen:
When creating new strategy and using default value. `ts_callback_rate` is set to 0 however if you exit then import strategy, `ts_callback_rate` value will become invalid and set to `None`. Returning value to 0 won't be accepted.

Test Summary:
 *Profit_taking feature:
  - create perpetual pmm strategy using ETH-USDT market pair
  - bid and ask spread set to 0.01%
  - set One-way mode
  - enable profit taking
  - long and short profit taking spread set to 0.6%
  - default value for other parameters
  - start strategy
  - wait for one side to get a fill: Ok ( bid order got a fill)
  - order refresh time and order creation are suspended after detecting open position
  - manually compute sell order price with respect to long_profit_taking_spread : computed price match with what's on the bot
  - set Hedge mode with the same configuration
  - limit bid order gets filled,  limit sell order is created with respect to long_profit_taking_spread : Ok 
  - limit sell order got a filled,  limit buy order is created with respect to short_profit_taking_spread : Ok  
	  
 *Trailing stop feature:	  
   - set bid spread to 0.01 while ask spread is set to 0.1
   - set position management to `trailing_stop` with `ts_activation_spread` = 1% and `ts_callback_rate` = 1%
   - start strategy: After one side gets fill, bot immediately place new order to prevent stop-loss

**Comment by vic-en** [comment]
> Tested PR and so far only one minor issue was seen:
> When creating new strategy and using default value. `ts_callback_rate` is set to 0 however if you exit then import strategy, `ts_callback_rate` value will become invalid and set to `None`. Returning value to 0 won't be accepted.
> 
> Test Summary:
> *Profit_taking feature:
> 
>     * create perpetual pmm strategy using ETH-USDT market pair
> 
>     * bid and ask spread set to 0.01%
> 
>     * set One-way mode
> 
>     * enable profit taking
> 
>     * long and short profit taking spread set to 0.6%
> 
>     * default value for other parameters
> 
>     * start strategy
> 
>     * wait for one side to get a fill: Ok ( bid order got a fill)
> 
>     * order refresh time and order creation are suspended after detecting open position
> 
>     * manually compute sell order price with respect to long_profit_taking_spread : computed price match with what's on the bot
> 
>     * set Hedge mode with the same configuration
> 
>     * limit bid order gets filled,  limit sell order is created with respect to long_profit_taking_spread : Ok
> 
>     * limit sell order got a filled,  limit buy order is created with respect to short_profit_taking_spread : Ok
> 
> 
> *Trailing stop feature:
> 
>     * set bid spread to 0.01 while ask spread is set to 0.1
> 
>     * set position management to `trailing_stop` with `ts_activation_spread` = 1% and `ts_callback_rate` = 1%
> 
>     * start strategy: After one side gets fill, bot immediately place new order to prevent stop-loss

Thanks, I fixed the issue when `ts_callback_rate` is left as default. Although, users would rarely leave it as 0.

Also, note when using the trailing stop feature that bot would only place new order to prevent stop-loss if the open position has a loss. You can confirm if that was the case and reason for the order that was closed immediately.

File: hummingbot/strategy/perpetual_market_making/perpetual_market_making_config_map.py [hunk 1]
```diff
@@ -186,6 +190,55 @@ def derivative_on_validated(value: str):
...
+        ConfigVar(key="ts_activation_spread",
+                  prompt="At what spread from the position entry price do you want the bot to start trailing? (Enter 1 for 1%) >>> ",
+                  required_if=lambda: perpetual_market_making_config_map.get("position_management").value == "Trailing_stop",
+                  type_str="decimal",
+                  default=Decimal("0"),
+                  validator=lambda v: validate_decimal(v, 0, 100, True),
+                  prompt_on_new=True),
+    "ts_callback_rate":
+        ConfigVar(key="ts_callback_rate",
+                  prompt="At what spread away from the trailing peak price do you want to positions to remain open before they're closed? (Enter 1 for 1%) >>> ",
```
Comment (vic-en) [inline hummingbot/strategy/perpetual_market_making/perpetual_market_making_config_map.py:227]:
```suggestion
                  prompt="At what spread away from the trailing peak price do you want positions to remain open before they're closed? (Enter 1 for 1%) >>> ",
```

**Review by Nullably — APPROVED** [review]
LGTM

**Comment by RC-13** [comment]
Confirm fix for the reported minor issue.
> Also, note when using the trailing stop feature that bot would only place new order to prevent stop-loss if the open position has a loss. You can confirm if that was the case and reason for the order that was closed immediately.
Yes, indeed, a new order will be immediately created to prevent further loss

**Review by RC-13 — APPROVED** [review]

**Pull request merged**